teams_accounts.go

```diff
@@ -0,0 +1,131 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"net/http"
+	"time"
+)
+
+type TeamsAccount struct {
+	GatewayID    string `json:"gateway_tag"`   // Internal teams ID
+	ProviderName string `json:"provider_name"` // Auth provider
+	CloudflareID string `json:"id"`            // cloudflare account ID
```

Comment:
```suggestion
	GatewayTag    string `json:"gateway_tag"`   // Internal teams ID
	ProviderName string `json:"provider_name"` // Auth provider
	ID string `json:"id"`            // cloudflare account ID
```

is there a reason the struct field is different to the JSON tag? generally we've used a 1:1 for simplicity and reduce confusion between the API and the SDK counterpart.

Reply:
Yeah. I changed the variable names to give a little more readability to the user. The original struct json tags we use in the API are confusing.

Reply:
i agree it can be confusing here, but i think we need to go one way or another as right now it's in limbo and hard to grok if you don't know the reasoning. i think we should rename the fields to match the JSON to avoid further confusion especially seeing how the API docs reference the JSON tags.